CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 
 # Options
-
+option(SLEEF_BUILD_SHARED_LIBS "Build sleef as static library" ON)
```

Comment:
Can you try this change instead and switch back to old name everywhere? I believe this can work according to the documentation (https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html)
```
-option(SLEEF_BUILD_SHARED_LIBS "Build sleef as static library" ON)
+# Declare BUILD_SHARED_LIBS option to make it accessible to top-level projects.
+option(BUILD_SHARED_LIBS "Build using shared libraries (CMake reserved)" ON)
+
```

Reply:
I have tried this method, It not constant work, the reason as above: https://github.com/shibatch/sleef/pull/510#issuecomment-1934336010

Reply:
pytorch default want setup “BUILD_SHARED_LIBS” as shared, but it want to use sleef as static library. Current pytorch is used a save/restore this variable to WA it. 
Save: https://github.com/pytorch/pytorch/blob/cee16353db384dd33a4804f9f26829cacfcfbab7/aten/src/ATen/CMakeLists.txt#L424-L440
Restore: https://github.com/pytorch/pytorch/blob/cee16353db384dd33a4804f9f26829cacfcfbab7/aten/src/ATen/CMakeLists.txt#L480-L485
And the original author think we should use SLEEF_ to pass the sleef build config also: https://github.com/pytorch/pytorch/blob/cee16353db384dd33a4804f9f26829cacfcfbab7/aten/src/ATen/CMakeLists.txt#L483